### DISCUSSION TITLE
Should the NDK and/or platform provide a shared libunwind?

### DISCUSSION THREAD

**Discussion by FelixZhang00** [android/ndk#1689] [Q&A]
a.so and b.so use a shared libunwind.so instead of static link libunwind.a

**Comment by DanAlbert** [comment]
The platform does provide libunwind.so. It's not available until API 29 (or maybe 30).

**Comment by rprichard** [comment]
The unwinder APIs are exposed from the platform's libc.so starting with API 30 (Android R).

https://android.googlesource.com/platform/bionic/+/refs/tags/android-11.0.0_r48/libc/libc.map.txt#1529

**Reply by FelixZhang00** [reply]
> The unwinder APIs are exposed from the platform's libc.so starting with API 30 (Android R).
> 
> https://android.googlesource.com/platform/bionic/+/refs/tags/android-11.0.0_r48/libc/libc.map.txt#1529

Can we(apk) provide libunwind.so like libc++_shared.so？

**Reply by DanAlbert** [reply]
To what end? Is statically linking libunwind.a taking up too much space in your APK?

**Reply by FelixZhang00** [reply]
> 

yes. we have many .so  statically linking libunwind.a

**Reply by DanAlbert** [reply]
You'll save even more space by turning all your shared libs into static libs and linking them all into a single shared lib.
